### PR TITLE
Prompt

### PR DESCRIPTION
**Pull request by dimostzim**

### CODE REVIEW

**Comment by MartinekV** [comment]
Success logging:
- Logging 'Level' is not clear, choose a better name for the metric
- There starts to be a repetition of code when logging -1 metrics on failure, extract this failure handling into a function and call that function instead when necessary

Prompting:
- Line 20, 22, 24 - we don't want these things out of our pipeline, we just want the model to validate the generalizability, not sure if that matters.
- Line 28 - We don't want to bias the model with extra information like this yet, keep the biology-specific stuff, but make it into a second prompt version
- Same with line 38, you can keep this, but make a minimal prompt first, and then extract all of these specifics into a separate prompt
- Line 47 - Keep stuff like this in the minimal prompt (absolute paths, directory instructions etc..). Line 51 should not be there.
- Line 57 - We don't need data processing or test_model files, those are optional, keep them for the more advanced prompt
- Line 75 - i believe smolagents already injects tool information and docs into the context, no need to repeat it
- Please somehow mark parts of the prompt that is re-used from smolagents prompt

**Comment by andrea-gariboldi** [comment]
All comments addressed in last two commits.

**Review by MartinekV — CHANGES_REQUESTED** [review]
Prompts are good. Logging needs minor fixes.

File: src/eval/evaluate_result.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import sys
 import pandas as pd
 from sklearn.metrics import average_precision_score, roc_auc_score
+from src.run_logging.evaluate_log_run import log_inference_stage
```
[the remark this answers is not in the document]
Comment (andrea-gariboldi) [reply src/eval/evaluate_result.py]: Added function to new file logging_helpers.py

File: src/eval/evaluate_result.py [hunk 2]
```diff
@@ -80,9 +81,7 @@ def evaluate_log_metrics(results_file, test_file, logging_fn, output_file=None,
     }
 
     # Log the metrics
-    logging_fn(metrics)
-
-    logging_fn({"Level": 2})
+    log_inference_stage(2, metrics=metrics)
```
[the remark this answers is not in the document]
Comment (andrea-gariboldi) [reply src/eval/evaluate_result.py]: Unified function log_inference_and_metrics.

**Comment by andrea-gariboldi** [comment]
Added the first version of BioPrompt.
